### ISSUE TITLE
Investigate why probe-rs cannot reset MIMXRT685 and inconsistent behavior coming out of reset

### ISSUE THREAD

**Issue by tullom** [OpenDevicePartnership/embassy-imxrt#201]

**Comment by jerrysxie** [comment]
@tullom Is this still an issue?

**Comment by tullom** [comment]
Will need to test on the RT685 EVK. The problem that sparked this issue was the inability to consistently run [embedded-test](https://docs.rs/embedded-test/0.7.0/embedded_test/) on the EVK. Work has been done on probe-rs since then so its worth it to give it another shot.

**Comment by tullom** [comment]
We were seeing a similar issue with teleprobe, so more investigation was done, and I think we found the problem. The IMXRT ROM code corrupts RAM when, and the ROM code runs every time we reset. `embedded-test` relies on this reset behaviour to run its tests, so that's why the tests were getting corrupted. The investigation results are beloew for future reference. I think we can close this issue now.

> TLDR: As of 05/19/2026, teleprobe tests can be executed from RAM on IMXRT.
> 
> At the beginning of the sprint, I tested running a simple "Hello World" app in RAM with teleprobe. It successfully compiled and ran to completion. I added some more logic and found that i would get strange errors, which varied from probe-rs not being able to find the RTT control block in memory to hard faulting at completely normal instructions. Only if the teleprobe test was trivial, it would successfully complete with no weirdness.
> 
> After consulting with @felipebalbi, he found that after reset, ROM code is corrupting regions of RAM we just wrote to with probe-rs. This is an inherent limitation of the chip, not caused by probe-rs or teleprobe. A small part of the beginning of RAM isnt impacted, which explains why smaller toy examples work but large examples do not.
> 
> Felipe turned his investigation into a fully blown tool, [rambo](https://github.com/felipebalbi/rambo), which can determine what areas of RAM get modified by RAM after the ROM code runs.
> 
> Since teleprobe calls probe-rs to reset the chip after writing each test to RAM, I wrote off the IMXRT board as not pursuable since almost the entirety of RAM gets corrupted on reset. For the rest of the sprint, I turned my sights onto the FRDM-MCXA266 board, which after using Felipe's tool, reports that only the first 12KiB of RAM gets touched by the ROM code. By defining RAM after the corrupted region, we get consistent non-corrupted test runs. At that point, I decided to choose the MCXA266 as the preferred HIL platform of choice.
> 
> Finally, Felipe put out a https://github.com/embassy-rs/teleprobe/pull/38 I tested the IMXRT board with the updated teleprobe and the problem is resolved! Thus, I have reverted back to using the IMXRT board as the preferred platform of choice.
